datafusion/sqllogictest/test_files/unnest.slt

```diff
@@ -0,0 +1,59 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+
+#   http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+############################
+# Unnest Expressions Tests #
+############################
+
+statement ok
+CREATE TABLE unnest_table
+AS VALUES
+    ([1,2,3]),
+    ([4,5]),
+    ([6])
+;
+
+query I
+select unnest([1,2,3]);
+----
+1
+2
+3
+
+query ?
+select unnest([]);
+----
+
+query ?
+select column1 from unnest_table;
+----
+[1, 2, 3]
+[4, 5]
+[6]
+
+query I
+select unnest(column1) from unnest_table;
+----
+1
+2
+3
+4
+5
+6
+
+statement ok
+drop table unnest_table;
```

Comment:
Awesome -- can you please also add tests for:
1.  to unnest two columns (it is fine to error, but I think that behavior should be tested)
2. trying to unnest a scalar value `1` rather than `[1]`
3. to unnest zero columns (again should be an error)